manager/integration/tests/test_node.py

```diff
@@ -2663,16 +2662,25 @@ def test_disk_eviction_with_node_level_soft_anti_affinity_disabled(client, # NOQ
             break
         time.sleep(common.RETRY_EXEC_INTERVAL)
     assert len(os.listdir(replica_path)) > 0
+    volume = wait_for_volume_healthy(client, volume_name)
 
     # Step 7
     replica_count = 1
-    volume = client.by_id_volume(vol_name)
-    volume = volume.updateReplicaCount(replicaCount=replica_count)
-    volume = common.wait_for_volume_healthy(client, vol_name)
+    volume.updateReplicaCount(replicaCount=replica_count)
+    volume = wait_for_volume_healthy(client, volume_name)
 
-    for r in volume.replicas:
-        if r.hostId != lht_hostId:
-            volume.replicaRemove(name=r.name)
+    for i in range(RETRY_COUNTS):
+        deleted = True
+        volume = client.by_id_volume(volume_name)
+        for r in volume.replicas:
+            if r.hostId != lht_hostId:
+                try:
+                    volume.replicaRemove(name=r.name)
+                except Exception:
+                    deleted = False
+        if deleted:
+            break
```

Comment:
Should also assert the volume has only 1 replica after the loop to ensure replicas got removed.

Reply:
Good idea! Yes the test can verify the only replica is the newly rebuilt replica then the following data r/w would validate the availability of the new replica.